src/java.base/share/classes/java/io/PrintStream.java

```diff
@@ -1261,12 +1408,17 @@ public PrintStream format(String format, Object ... args) {
      */
     public PrintStream format(Locale l, String format, Object ... args) {
         try {
-            synchronized (this) {
-                ensureOpen();
-                if ((formatter == null)
-                    || (formatter.locale() != l))
-                    formatter = new Formatter(this, l);
-                formatter.format(l, format, args);
+            if (lock != null) {
+                lock.lock();
+                try {
+                    implFormat(l, format, args);
+                } finally {
+                    lock.unlock();
+                }
+            } else {
+                synchronized (this) {
+                    implFormat(format, args);
```

Comment:
I think there's a typo here. I think it should have been:

```
implFormat(l, format, args);
```

Reply:
Well spotted, the locale should be provided.